test/index_spec.js

```diff
@@ -174,4 +177,31 @@ describe('graphqlFields', () => {
                 done();
             }).catch(done);
     });
+
+    it('should use field alias when it is defined in the query', function (done) {
+        const query = `
+            query UsersRoute {
+                viewer {
+                    users(userId:"123",first:25,includeInactive:true) @skip(if:false) {
+                        pageInfo {
+                            total:totalResults
+                        }
+                    }
+                }
+            }
+        `;
+
+        graphql.graphql(schema, query, null, {})
+            .then(() => {
+                const expected = {
+                    users: {
+                        pageInfo: {
+                            total: {}
```

Comment:
I'm not sure I agree with this change. This library is used to determine which fields the client wants to resolve on the server. Since the client can rename any field using an alias you would not be able to determine the actual underlining field that was requested if we updated the returned object to be the alias.

Reply:
Makes sense. Closed the PR as I found a different solution to my problem :)